pkg/aggregator/mocksender/asserts.go

```diff
@@ -14,64 +14,78 @@ import (
+	return okCall && notOkCalls
 }
 
 // AssertMetric allows to assert a metric was emitted with given parameters.
 // Additional tags over the ones specified don't make it fail
 func (m *MockSender) AssertMetric(t *testing.T, method string, metric string, value float64, hostname string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, value, hostname, AssertTagsContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, value, hostname, AssertTagsContains(tags))
 }
 
 // AssertMetricInRange allows to assert a metric was emitted with given parameters, with a value in a given range.
 // Additional tags over the ones specified don't make it fail
 func (m *MockSender) AssertMetricInRange(t *testing.T, method string, metric string, min float64, max float64, hostname string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, AssertFloatInRange(t, min, max), hostname, AssertTagsContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, AssertFloatInRange(min, max), hostname, AssertTagsContains(tags))
 }
 
 // AssertMetricTaggedWith allows to assert a metric was emitted with given tags, value and hostname not tested.
 // Additional tags over the ones specified don't make it fail
 func (m *MockSender) AssertMetricTaggedWith(t *testing.T, method string, metric string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsContains(tags))
 }
 
 // AssertMetricNotTaggedWith allows to assert tags were never emitted for a metric.
 func (m *MockSender) AssertMetricNotTaggedWith(t *testing.T, method string, metric string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsNotContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsNotContains(tags))
+}
+
+// Return a bool value if all the elt of expected are inside the actual array
+func expectedInActual(expected, actual []string) bool {
+	expectedCount := 0
+	for _, e := range expected {
+		for _, a := range actual {
+			if e == a {
+				expectedCount++
+				break
+			}
+		}
+	}
+	return len(expected) == expectedCount
+}
+
+// AnythingBut match everything except the argument
+func AnythingBut(expected interface{}) interface{} {
+	return mock.MatchedBy(func(actual interface{}) bool {
+		return !assert.ObjectsAreEqualValues(expected, actual)
+	})
 }
 
 // AssertTagsContains is a mock.argumentMatcher builder to be used in asserts.
 // It allows to check if tags are emitted, ignoring unexpected ones and order.
-func AssertTagsContains(t *testing.T, expected []string) interface{} {
+func AssertTagsContains(expected []string) interface{} {
```

Comment:
We should find a better name than assert*, maybe MatchTagsContains?
same for all the others